templates/single-product/add-to-cart/subscription.php

```diff
@@ -39,16 +38,18 @@
 		<?php
 		do_action( 'woocommerce_before_add_to_cart_quantity' );
 
-		woocommerce_quantity_input( array(
-			'min_value'   => apply_filters( 'woocommerce_quantity_input_min', $product->get_min_purchase_quantity(), $product ),
-			'max_value'   => apply_filters( 'woocommerce_quantity_input_max', $product->get_max_purchase_quantity(), $product ),
-			'input_value' => isset( $_POST['quantity'] ) ? wc_stock_amount( wp_unslash( $_POST['quantity'] ) ) : $product->get_min_purchase_quantity(), // WPCS: CSRF ok, input var ok.
-		) );
+		woocommerce_quantity_input(
+			[
+				'min_value'   => apply_filters( 'woocommerce_quantity_input_min', $product->get_min_purchase_quantity(), $product ),
+				'max_value'   => apply_filters( 'woocommerce_quantity_input_max', $product->get_max_purchase_quantity(), $product ),
+				'input_value' => isset( $_POST['quantity'] ) ? wc_stock_amount( wp_unslash( $_POST['quantity'] ) ) : $product->get_min_purchase_quantity(), // phpcs:ignore WordPress.Security.NonceVerification.Missing -- input var ok.
+			]
+		);
 
 		do_action( 'woocommerce_after_add_to_cart_quantity' );
 		?>
 
-		<button type="submit" class="single_add_to_cart_button button alt wp-element-button" name="add-to-cart" value="<?php echo esc_attr( $product->get_id() ); ?>"><?php echo esc_html( $product->single_add_to_cart_text() ); ?></button>
+		<button type="submit" class="single_add_to_cart_button button alt<?php echo esc_attr( wc_wp_theme_get_element_class_name( 'button' ) ? ' ' . wc_wp_theme_get_element_class_name( 'button' ) : '' ); ?>" name="add-to-cart" value="<?php echo esc_attr( $product->get_id() ); ?>"><?php echo esc_html( $product->single_add_to_cart_text() ); ?></button>
```

Comment:
```suggestion
		<button type="submit" class="single_add_to_cart_button button alt<?php echo wc_wp_theme_get_element_class_name( 'button' ) ? ' ' . esc_attr( wc_wp_theme_get_element_class_name( 'button' ) : '' ); ?>" name="add-to-cart" value="<?php echo esc_attr( $product->get_id() ); ?>"><?php echo esc_html( $product->single_add_to_cart_text() ); ?></button>
```

Happy to approve without this change, but this caught my eye. The way this is written, the whole ternary is inside the `esc_attr()` call and so if the condition is false, it will escape an empty string ('') or a space plus the class `'  wp-element-button'`. While this is doesn't cause any issues, it didn't seem right to me. 

The proposed change just escapes the variable part of this, the `wc_wp_theme_get_element_class_name()` output.

Reply:
Thanks @james-allan for the review. I thought this too, but then I noticed WooCommerce core code escapes the whole ternary everywhere they call this function. I can't think of a reason why, but I followed suit 😄 

The suggested code has a syntax issue due to the closing `)` for the `esc_attr` not being correct, so I think I'll just leave it for now in favour of keeping it the same as WooCommerce core.